CribblyBackend/Program.cs

```diff
@@ -1,23 +1,43 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Configuration;
+using Serilog;
 
 namespace CribblyBackend
 {
     public class Program
     {
+        private const string _startup = "----------------------Application Starting----------------------";
+        private const string _shutdown = "----------------------Application Stopping----------------------";
         public static void Main(string[] args)
         {
-            CreateHostBuilder(args).Build().Run();
+            var configuration = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json")
+                .Build();
+            //Configure Serilog
+            Log.Logger = new LoggerConfiguration()
+                .ReadFrom.Configuration(configuration)
+                .CreateLogger();
```

Comment:
Instead of using the static methods available on `Log`, would it be better to inject an `ILogger` wherever we need it? I _think_ that's what we'd need to do if we wanted to mock the logger during unit testing and ensure that it's getting called.

Reply:
After doing some research, this is accomplished by adding the following to `ConfigureServices` in `Startup.cs`:

```cs
// ... other service config
services.AddSingleton(Log.Logger);
// ...
```
And then you can inject and call the logger. For example, in `TeamController.cs`:
```cs
// ... other usings
using Serilog;
// ... other usings

namespace CribblyBackend.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class TeamController : ControllerBase
    {
        private readonly ITeamService teamService;
        private readonly ILogger logger; // <-- new
        public TeamController(ITeamService teamService, ILogger logger) // <-- second arg is new
        {
            this.teamService = teamService;
            this.logger = logger; // <-- new
        }
        // now you can call logger.Debug instead of Log.Debug everywhere in the controller
    }
}
```

Reply:
> Instead of using the static methods available on `Log`, would it be better to inject an `ILogger` wherever we need it? I _think_ that's what we'd need to do if we wanted to mock the logger during unit testing and ensure that it's getting called.

I can register it in `startup.cs` and inject, but the reason I am calling the logger in `Program.cs` is so that errors can be caught and logged if the program fails to start. If you don't think we need that though, we can take it out

Reply:
Yeah that's fine for `Program.cs`, I just meant to inject it everywhere else (like controllers and services)